### PR TITLE
Refactor operations to fully work on AArch64.

### PR DESCRIPTION
**Pull request by diegorusso**
Create IntelPstateOperation class for Intel P-State operations.
Enable CPUGovernor on AArch64 platforms.
Replace all Class names with super() in the __init__ method.

### CODE REVIEW

**Comment by diegorusso** [comment]
cc @vstinner @corona10 @mdboom

**Comment by corona10** [comment]
I will take a look by this weekend :)

**Review by vstinner — COMMENTED** [review]
Can you try to avoid moving code? It makes the diff harder to review.

Like leaving TurboBoostIntelPstate after TurboBoostMSR.

File: pyperf/_system.py [hunk 1]
```diff
@@ -119,18 +120,113 @@ def write(self, tune):
...
+
+        msg = "%r written into %s" % (content, self.path)
+        action = 'enabled' if enable else 'disabled'
+        self.log_action("Turbo Boost %s: %s" % (action, msg))
+
+
+class CheckNOHZFullIntelPstate(IntelPstateOperation):
+
+    def __init__(self, system):
+        IntelPstateOperation.__init__(self, 'Check nohz_full', system)
```
Comment (corona10) [inline pyperf/_system.py:199]: nit
```suggestion
        super.__init__('Check nohz_full', system)
```

File: pyperf/_system.py [hunk 2]
```diff
@@ -119,18 +120,113 @@ def write(self, tune):
...
+
+
+class TurboBoostIntelPstate(IntelPstateOperation):
+    """
+    Get/Set Turbo Boost mode of Intel CPUs by reading from/writing into
+    /sys/devices/system/cpu/intel_pstate/no_turbo of the intel_pstate driver.
+    """
+
+    def __init__(self, system):
+        IntelPstateOperation.__init__(self, 'Turbo Boost (intel_pstate)', system)
```
Comment (corona10) [inline pyperf/_system.py:136]:
```suggestion
        super.__init__('Turbo Boost (intel_pstate)', system)
```

File: pyperf/_system.py [hunk 3]
```diff
@@ -266,125 +362,92 @@ def write(self, tune):
...
-                            'CPU frequency')
-
-    def write(self, tune):
-        enable = (not tune)
-
-        self.read_turbo_boost()
-        if self.enabled == enable:
-            # no_turbo already set to the expected value
-            return
+        Operation.__init__(self, 'CPU scaling governor', system)
```
Comment (corona10) [inline pyperf/_system.py:378]: ditto

File: pyperf/_system.py (hunk 2 above)
Comment (vstinner) [reply pyperf/_system.py:136]: It's `super()` with parens, no?

File: pyperf/_system.py [hunk 4]
```diff
@@ -950,18 +984,21 @@ def write(self, tune):
 
 
 OPERATIONS = [
+    # Generic operations
     PerfEvent,
     ASLR,
     LinuxScheduler,
     CPUFrequency,
+    IRQAffinity,
+    PowerSupply,
```
Comment (vstinner) [inline pyperf/_system.py:993]: Add an empty line for readability. Same for the other categories below.

File: pyperf/_system.py (hunk 1 above)
Comment (diegorusso) [reply pyperf/_system.py]: Addressed your comment

File: pyperf/_system.py (hunk 2 above)
Comment (diegorusso) [reply pyperf/_system.py]: Addressed your comment

File: pyperf/_system.py (hunk 4 above)
Comment (diegorusso) [reply pyperf/_system.py:993]: I hope I understood what you mean, can you check?

File: pyperf/_system.py [hunk 5]
```diff
@@ -119,21 +120,23 @@ def write(self, tune):
...
+class IntelPstateOperation(Operation):
+    @staticmethod
+    def available():
+        return use_intel_pstate()
+
+
 class TurboBoostMSR(Operation):
     """
-    Get/Set Turbo Boost mode of Intel CPUs using /dev/cpu/N/msr.
+    Get/Set Turbo Boost mode of X86 CPUs using /dev/cpu/N/msr
```
Comment (vstinner) [inline pyperf/_system.py:131]:
```suggestion
    Get/Set Turbo Boost mode of x86 CPUs using /dev/cpu/N/msr
```

File: pyperf/_system.py (hunk 4 above)
Comment (vstinner) [reply pyperf/_system.py:993]: Yes, thank you.

**Review by vstinner — APPROVED** [review]
The overall change LGTM. I let @corona10 do a second  review.

**Review by corona10 — APPROVED** [review]

**Pull request merged**